### ISSUE TITLE
[Arm64/Unix] Crossgen NuGet.Protocol.Core.v3.dll segfaults

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#10361]
Crossgen NuGet.Protocol.Core.v3.dll segfaults

Issue seems to be that in frame 4 MethodTable::GetMethodDescForSlotAddress(pTgt) is called with a null pTgt.

```
#0  Precode::GetType (this=0x0) at /home/sdmaclea/git/coreclr/src/vm/crossgen/../precode.h:128
#1  0x000000000070ca58 in Precode::GetPrecodeFromEntryPoint (addr=0, fSpeculative=0) at /home/sdmaclea/git/coreclr/src/vm/crossgen/../precode.h:279
#2  0x000000000070a0cc in MethodDesc::GetMethodDescFromStubAddr (addr=0, fSpeculative=0) at /home/sdmaclea/git/coreclr/src/vm/method.cpp:4386
#3  0x0000000000712e1c in MethodTable::GetMethodDescForSlotAddress (addr=0, fSpeculative=0) at /home/sdmaclea/git/coreclr/src/vm/methodtable.cpp:7544
#4  0x0000000000712c5c in MethodTable::GetMethodDescForInterfaceMethod (this=0x7fb582b700, ownerType=..., pInterfaceMD=0x7fb59e6830) at /home/sdmaclea/git/coreclr/src/vm/methodtable.cpp:2136
#5  0x00000000006eea60 in resolveVirtualMethodHelper (callerMethod=0x7fb59a64f8, baseMethod=0x7fb59e6830, derivedClass=0x7fb582b700, ownerType=0x7fb4ec1bf1) at /home/sdmaclea/git/coreclr/src/vm/jitinterface.
```

**Comment by sdmaclea** [comment]
@AndyAyersMS Is this related to the recent interface changes

**Comment by AndyAyersMS** [comment]
Seems likely. 

Might be another instance of #10131, which I'm testing a fix for now.

**Comment by AndyAyersMS** [comment]
Ooops, meant #10311.

**Comment by AndyAyersMS** [comment]
Let me know if you can still repro this now that a fix for #10311 is in (via #10371)

**Comment by sdmaclea** [comment]
Fixed by #10371
